src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners.xml

```diff
@@ -103,8 +105,8 @@
 
         <service id="Sylius\Bundle\CoreBundle\EventListener\ProductDeletionListener">
             <argument type="service" id="request_stack" />
-            <argument type="service" id="Sylius\Component\Core\Promotion\Updater\Rule\ContainsProductRuleUpdater" />
-            <tag name="kernel.event_listener" event="sylius.product.post_delete" method="removeProductFromPromotionRules" />
```

Comment:
As you removed the `removeProductFromPromotionRules` how the removal of product is handled now?

Reply:
We are not allowed to remove the product if it is assigned to promotion rule, so it is just prevented from removing

Reply:
Ah, right. Thanks.